Sage/carbon
# fix(input): ensure focus styles removed when becoming inert

Our input components manage focus styles using react state rather than CSS, due to limitations in their implementation. This state is changed in the input's focus and blur events. However when opening a modal, we add the `inert` attribute to all parts of the DOM tree outside the modal - this causes the input to lose focus, as inert elements are not supposed to receive focus in any descendants. In this situation, not all browsers fire the `blur` event on the input that's losing focus - in particular, Firefox does not. This results in the focus style remaining on the now-unfocused input. To fix this we need to manually call the `blur` method on any currently-focused element before adding the `inert` attribute.

fix #6415

### Proposed behaviour

Focus styles should never be applied to a Select, Textbox or any other input component once it has lost focus



### Current behaviour

In the Firefox browser, when opening a dialog with an input focused, the focus style remains on the input even though it is no longer focused



### Checklist




#### QA


### Additional context



### Testing instructions



Test [codesandbox linked in the Github issue](https://codesandbox.io/s/flamboyant-bhabha-t6t9y5?file=/src/App.js) in Firefox and ensure that the issue is fixed. Test in all supported browsers to ensure no similar focus issues occur in any of them.
